bootstrapper/src/utils/mod.rs

```
@@ -54,20 +59,29 @@ pub async fn wait_for_transaction(
 
     let transaction_status = transaction_receipt.ok().unwrap();
 
-    match transaction_status {
-        TransactionReceiptWithBlockInfo { receipt: TransactionReceipt::Invoke(receipt), .. } => {
-            log::trace!("txn : {:?} : {:?}", tag, receipt);
-        }
-        TransactionReceiptWithBlockInfo { receipt: TransactionReceipt::DeployAccount(receipt), .. } => {
+    log::trace!("txn : {:?} : {:?}", tag, transaction_status);
+    let exec_result: ExecutionResult = match transaction_status.receipt {
+        TransactionReceipt::Invoke(receipt) => receipt.execution_result,
+        TransactionReceipt::DeployAccount(receipt) => {
             let contract_address = receipt.contract_address;
-            log::trace!("txn : {:?} : {:?}", tag, contract_address);
+            log::info!("Account deployed at {:?}", contract_address);
+            receipt.execution_result
         }
-        _ => {
-            log::error!("Transaction status: {:?}", transaction_status);
-            panic!("Transaction failed");
+        TransactionReceipt::Declare(receipt) => receipt.execution_result,
+        TransactionReceipt::Deploy(receipt) => {
+            log::info!("Tag: {:?}, Contract deployed at address {:?}", tag, receipt.contract_address);
+            receipt.execution_result
         }
+        TransactionReceipt::L1Handler(receipt) => receipt.execution_result,
     };
 
+    match exec_result {
+        ExecutionResult::Succeeded => {}
+        ExecutionResult::Reverted { reason } => {
+            panic!("Transaction failed with {:?}", reason);
```

do we need to panic just here tho? we can return an error right?

---

Since bootstrapper works like a script is not capable enough to retry a txn, also can you continue from an already run state. So I thought panicking should be fine